xonsh/built_ins.py

```diff
@@ -98,6 +98,34 @@ def expand_path(s):
     return os.path.expanduser(s)
 
 
+def expand_braces(s):
+    # XXX: Doesn't handle escaping literal braces
+    """Takes a string and expands brace-enclosed comma-separated lists into
+    space-separated arguments.
+    """
+    regex = re.compile('(.*?){(.*)}(.*)')
+    def _split_braces(s, nesting=0):
```

Comment:
I think you can define this function globally, also you can define the regex globally and lazily